src/main/java/frc/states/MechState.java

```diff
@@ -0,0 +1,15 @@
+package frc.states;
+
+public interface MechState {
```

Comment:
I think if you declare this as an abstract base class, then you'll save yourself some code. You get two benefits:

* First, the abstract base class can provide default implementations for any method. So you can default to methods that do nothing. This means then in all the derived classes, if you don't need to do something special in a given function, you simply don't need to implement it.

* Second, an abstract base class can have fields. Seems like all the derived classes have identical fields, so you could hoist these fields into the abstract class.

Reply:
I've changed MechState to an abstract base class